dataset/dataset.go

```diff
@@ -48,6 +49,20 @@ type Client struct {
 	url string
 }
 
+// closeResponseBodyWithContext closes the response body and logs an error containing the context if unsuccessful
+func closeResponseBodyWithContext(ctx context.Context, resp *http.Response) {
+	if err := resp.Body.Close(); err != nil {
+		log.ErrorCtx(ctx, err, log.Data{"Message": "error closing http response body"})
+	}
+}
+// closeResponseBody closes the response body and logs an error if unsuccessful
+
+func closeResponseBody(resp *http.Response) {
```

Comment:
do we really need the second function? as it's only used one, I'd suggest only having `func closeResponseBody(ctx, resp)` and just passing nil context in when it's not available? or `context.Background()`